src/core/iterstate.rs

```diff
@@ -368,19 +369,23 @@ mod tests {
         assert_eq!(state.get_prev_param(), param);
         assert_eq!(state.get_best_param(), param);
         assert_eq!(state.get_prev_best_param(), param);
-        assert_eq!(state.get_cost(), std::f64::INFINITY);
-        assert_eq!(state.get_prev_cost(), std::f64::INFINITY);
-        assert_eq!(state.get_best_cost(), std::f64::INFINITY);
-        assert_eq!(state.get_prev_best_cost(), std::f64::INFINITY);
-        assert_eq!(state.get_target_cost(), std::f64::NEG_INFINITY);
+
+        assert!(state.get_cost().is_infinite());
+        assert!(state.get_prev_cost().is_infinite());
+        assert!(state.get_best_cost().is_infinite());
+        assert!(state.get_prev_best_cost().is_infinite());
+        assert!(state.get_target_cost().is_infinite());
```

Comment:
These tests make sure that the initial values are set correctly. Unfortunately, `.is_infinite()` does not consider the sign. For instance, it is important that `target_cost` is initialized with `NEG_INFINITY` and not `INFINITY`.

Reply:
Is the sign relevant for all of them? It is possible to do direct byte checks here, but it would be somewhat uncommon to see that, which is why I opted for `is_finite()` checks. 

I can replace these with:

```rust
assert_eq!(state.get_target_cost().to_ne_bytes(), f64::NEG_INFINITY.to_ne_bytes());
```

Or positive infinity if that is relevant to the test itself.

Reply:
Yes, the sign is relevant for all of them. What do you think about asserting a combination of `.is_finite()` and `.is_sign_positive()` or `.is_sign_negative()`? I believe this would most clearly show what the intention is.

Reply:
That's actually a fantastic idea, and preserves the intent. I should have checked to see a predicate for signage as well.

I will push a fixup commit up for this tomorrow morning.

Reply:
See c163b362812fa96c2000502c293ce3d2823d4648. 

If the changes are good to go, let me know and I'll do an autosquash to wrap this back into the original commit.